src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -707,181 +711,181 @@ private void InitFromStreamWithSize(Stream stream, int width, int height)
+                        }
                     }
                 }
-            }
 
-            //now read in the icon data
-            for (int j = 0; j < dirEntryCount; j++)
-            {
-                // process ignored into IconDump
-                if (iconDir.idEntries[j].ignore)
+                // Now read in the icon data
+                bool valid = false;
+                for (int j = 0; j < dirEntryCount; j++)
                 {
-                    IconDump id = new IconDump();
                     stream.Seek(iconDir.idEntries[j].imageOffset, SeekOrigin.Begin);
-                    id.data = new byte[iconDir.idEntries[j].bytesInRes];
-                    stream.Read(id.data, 0, id.data.Length);
-                    imageData[j] = id;
-                    continue;
-                }
-                // standard image
-                IconImage iidata = new IconImage();
-                BitmapInfoHeader bih = new BitmapInfoHeader();
-                stream.Seek(iconDir.idEntries[j].imageOffset, SeekOrigin.Begin);
-                byte[] buffer = new byte[iconDir.idEntries[j].bytesInRes];
-                stream.Read(buffer, 0, buffer.Length);
-                BinaryReader bihReader = new BinaryReader(new MemoryStream(buffer));
-                bih.biSize = bihReader.ReadUInt32();
-                bih.biWidth = bihReader.ReadInt32();
-                bih.biHeight = bihReader.ReadInt32();
-                bih.biPlanes = bihReader.ReadUInt16();
-                bih.biBitCount = bihReader.ReadUInt16();
-                bih.biCompression = bihReader.ReadUInt32();
-                bih.biSizeImage = bihReader.ReadUInt32();
-                bih.biXPelsPerMeter = bihReader.ReadInt32();
-                bih.biYPelsPerMeter = bihReader.ReadInt32();
-                bih.biClrUsed = bihReader.ReadUInt32();
-                bih.biClrImportant = bihReader.ReadUInt32();
-                iidata.iconHeader = bih;
-                //Read the number of colors used and corresponding memory occupied by
-                //color table. Fill this memory chunk into rgbquad[]
-                int numColors;
-                switch (bih.biBitCount)
-                {
-                    case 1:
-                        numColors = 2;
-                        break;
-                    case 4:
-                        numColors = 16;
-                        break;
-                    case 8:
-                        numColors = 256;
-                        break;
-                    default:
-                        numColors = 0;
-                        break;
-                }
+                    byte[] buffer = new byte[iconDir.idEntries[j].bytesInRes];
+                    stream.Read(buffer, 0, buffer.Length);
+                    using (BinaryReader bihReader = new BinaryReader(new MemoryStream(buffer)))
+                    {
+                        uint headerSize = bihReader.ReadUInt32();
```

Comment:
mixture of uint and int ?

Reply:
What do you mean? I'm declaring an uint variable and reading UInt32?

Reply:
disregard 😛